contract/assembly/marketplace.ts

```diff
@@ -0,0 +1,33 @@
+import { context, ContractPromiseBatch, storage, u128 } from "near-sdk-as";
+import { Chunk, CHUNK_SIZE } from "./model";
+
+export function sellChunk_impl(x: u32, y: u32, price: u128): void {
+    const chunk_location = getChunkLocationByGlobalCoords(x, y);
+    const chunk = Chunk.get(chunk_location.x, chunk_location.y);
+    chunk.price = price;
+    storage.set(Chunk.key(chunk_location.x, chunk_location.y), chunk);
+}
+
+export function buyChunk_impl(x: u32, y: u32): void {
+    const chunk_location = getChunkLocationByGlobalCoords(x, y);
```

Comment:
not sure why would you pass global coords here

like my assumption is that on marketplace you operate on chunk level (i.e. don't get inside of one chunk), so I'm a bit confused

Reply:
actually to be more precise, chunk currently is 16x16 – but I'd like to have 64x64 pieces of land tradable

so need to introduce separate larger entity that gets traded instead of trading chunks directly